apps/prairielearn/src/pages/instructorInstanceAdminSettings/instructorInstanceAdminSettings.html.ts

```diff
@@ -0,0 +1,278 @@
+                                buttonID: 'changeCiidButton',
+                                id_old: resLocals.course_instance.short_name,
+                                ids: shortNames,
+                                changeIdFormHelpText:
+                                  'The recommended format is <code>Fa19</code> or <code>Fall2019</code>. Add suffixes if there are multiple versions, like <code>Fa19honors</code>.',
+                              },
+                            )}"
+                            data-trigger="manual"
+                            onclick="$(this).popover('show')"
+                          >
+                            <i class="fa fa-i-cursor"></i>
+                            <span>Change CIID</span>
+                          </button>
+                        `
+                      : ''}
+                  </td>
+                </tr>
+                <tr>
+                  <th>Configuration</th>
+                  <td>
+                    ${EditConfiguration({
+                      hasCoursePermissionView: resLocals.authz_data.has_course_permission_view,
+                      hasCoursePermissionEdit: resLocals.authz_data.has_course_permission_edit,
+                      exampleCourse: resLocals.course.example_course,
+                      urlPrefix: resLocals.urlPrefix,
+                      navPage: resLocals.navPage,
+                      infoCourseInstancePath,
+                    })}
+                  </td>
+                </tr>
+                <tr>
+                  <th class="align-middle">Student Link</th>
+                  <td class="form-inline align-middle">
+                    <span class="input-group">
+                      <span readonly class="form-control form-control-sm">${studentLink}</span>
+                      <div class="input-group-append">
+                        <button
+                          type="button"
+                          class="btn btn-sm btn-outline-secondary btn-copy"
+                          data-clipboard-text="${studentLink}"
+                          aria-label="Copy student link"
+                        >
+                          <i class="far fa-clipboard"></i>
+                        </button>
+                        <button
+                          type="button"
+                          title="Student Link QR Code"
+                          aria-label="Student Link QR Code"
+                          class="btn btn-sm btn-outline-secondary js-student-link-qrcode-button"
+                        >
+                          <i class="fas fa-qrcode"></i>
+                        </button>
+                      </div>
+                    </span>
+                    <div class="d-none">
+                      <div id="js-student-link-qrcode">
+                        <center>${resLocals.studentLinkQRCode}</center>
+                      </div>
+                    </div>
```

Comment:
This doesn't work correctly:

<img width="269" alt="Screenshot 2024-04-02 at 09 58 42" src="https://github.com/PrairieLearn/PrairieLearn/assets/1476544/49cd3b36-5342-4111-a474-fdd367c0baa1">

You probably need an `unsafeHtml(...)` wrapping the `resLocals.studentLinkQRCode` to ensure that it doesn't get escaped.

Reply:
Added `unsafeHtml(...)` here and QR code is now working in [9257025](https://github.com/PrairieLearn/PrairieLearn/pull/9673/commits/9257025e843a376237a056af7c027acc5002c911).